examples/ann/annoy_nsmlib_example.ipynb

```diff
@@ -0,0 +1,725 @@
+      "User 450\n",
+      "     Known positives:\n",
+      "        Kolya (1996)\n",
+      "        Devil's Own, The (1997)\n",
+      "        Contact (1997)\n",
+      "     Recommended:\n",
+      "        Murder at 1600 (1997)\n",
+      "        Scream (1996)\n",
+      "        Cop Land (1997)\n"
+     ]
+    }
+   ],
+   "source": [
+    "sample_recommendation_original(model, movielens, [3, 25, 450])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So pretty similar! How about speed?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "100 loops, best of 3: 3.85 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "sample_recommendation_original(model, movielens, [3, 25, 450], print_output=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The slowest run took 5.66 times longer than the fastest. This could mean that an intermediate result is being cached.\n",
+      "100 loops, best of 3: 2.55 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%timeit\n",
+    "sample_recommendation([3,25,450], model, movielens, print_output=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So a small difference at thes volumes, but let's see how it compares to an sklearn distance method as n increases:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def recommend_item_for_item(item_vecotrs):\n",
+    "    distances = pairwise_distances(item_vectors[0].reshape(1, -1), Y=item_vectors)\n",
+    "    top_items = distances.argsort()[:10]\n",
+    "    return top_items\n",
+    "\n",
+    "def time_recommend_item_for_item():\n",
+    "    start = time.time()\n",
+    "    recommend_item_for_item(item_vectors)\n",
+    "    end = time.time()\n",
+    "    return end - start\n",
+    "\n",
+    "def time_Annoy_recommend_item_for_item():\n",
+    "    start = time.time()\n",
+    "    temp_t.get_nns_by_vector(item_vectors[0], 10)\n",
+    "    end = time.time()\n",
+    "    return end - start\n",
+    "\n",
+    "def time_nms_recommend_item_for_item():\n",
+    "    start = time.time()\n",
+    "    nms_idx.knnQuery(item_vectors[0])[0]\n",
+    "    end = time.time()\n",
+    "    return end - start\n",
+    "\n",
+    "Annoy_response_times = []\n",
+    "pairwise_response_times = []\n",
+    "nms_response_times = []\n",
+    "\n",
+    "f = item_vectors.shape[1]\n",
+    "\n",
+    "for i in range (1, 500, 50):\n",
+    "    \n",
+    "    #Pairwise\n",
+    "    \n",
+    "    data = np.repeat(item_vectors, i, axis=0)   \n",
```

Comment:
Could try to use `numpy.broadcast_to` (see [this SO anwser](https://stackoverflow.com/a/37819585/1791279))

Reply:
I can only see this used with 1d arrays? Looks like a useful function.